### PR TITLE
pmem: Add namespace and namespace_append options

### PR DESCRIPTION
**Pull request by mizumm**
##### SUMMARY

Add namespace and namespace_append options to pmem module.

- namespace: Configure the namespace of PMem
- namespace_append: Enables to append the new namespaces.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

pmem

##### ADDITIONAL INFORMATION

```yaml
- name: Configure the two namespaces.
  community.general.pmem:
    namespace:
      - size: 1GiB
        type: pmem
        mode: raw
      - size: 320MiB
        type: pmem
        mode: sector
```

### CODE REVIEW

File: plugins/modules/storage/pmem/pmem.py [hunk 1]
```diff
@@ -76,6 +76,41 @@
...
+      type:
+        description:
+         - The type of namespace. The detail of the type is in the man page of ndctl-create-namespace.
+        type: str
+        required: false
+        choices: ['pmem', 'blk']
+      size:
+        description:
+          - The size of namespace. This option supports the suffixes "k" or "K" for KiB,
+            "m" or "M" for MiB, "g" or "G" for GiB and "t" or "T" for TiB.
```
Comment (felixfontein) [inline plugins/modules/storage/pmem/pmem.py:101]:
```suggestion
          - The size of namespace. This option supports the suffixes C(k) or C(K) for KiB,
            C(m) or C(M) for MiB, C(g) or C(G) for GiB and C(t) or C(T) for TiB.
```

File: plugins/modules/storage/pmem/pmem.py [hunk 2]
```diff
@@ -104,6 +140,9 @@
         socket:
           description: The socket ID to be configured.
           type: int
+        namespace:
+          description: The list of the detail of namespace.
+          type: list
```
Comment (felixfontein) [inline plugins/modules/storage/pmem/pmem.py:145]:
```suggestion
          type: list
          version_added: 4.6.0
```

File: plugins/modules/storage/pmem/pmem.py [hunk 3]
```diff
@@ -248,7 +321,96 @@ def pmem_is_dcpmm_installed(self):
...
+        types = []
+        for rg in region:
+            if rg['type'] not in types:
+                types.append(rg['type'])
+
+        return types
+
     def pmem_argument_check(self):
+        def convert_to_bytes(size_str):
+            size = int(size_pattn.match(size_str).group(1))
```
Comment (felixfontein) [inline plugins/modules/storage/pmem/pmem.py:349]: Why not simply use `module.human_to_bytes()` to convert strings to amounts of bytes?

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

File: plugins/modules/storage/pmem/pmem.py (hunk 3 above)
Comment (mizumm) [reply plugins/modules/storage/pmem/pmem.py]: Thank you for letting me know about human_to_bytes(). It looks great! I have applied the change to use the API.

File: plugins/modules/storage/pmem/pmem.py [hunk 4]
```diff
@@ -76,6 +76,41 @@
...
+          - The size of namespace. This option supports the suffixes C(k) or C(K) or C(KB) for KiB,
+            C(m) or C(M) or C(MB) for MiB, C(g) or C(G) or C(GB) for GiB and C(t) or C(T) or C(TB) for TiB.
+          - This option is required if multiple namespaces are configured.
+          - If this option is not set, all of the avaiable space of a region is configured.
+        type: str
+        required: false
+  namespace_append:
+    description:
+     - Enable to append the new namespaces to the system.
+     - The default is false so the all existed namespaces are removed by default.
```
Comment (felixfontein) [inline plugins/modules/storage/pmem/pmem.py:109]:
```suggestion
     - The default is C(false) so the all existing namespaces not listed in I(namespaces) are removed.
```

**Review by felixfontein — COMMENTED** [review]
Since the 4.5.0 release will only happen on Tuesday, I would suggest to remove the changelog fragment and the `version_added: 4.6.0` so we can merge this before the release.

File: plugins/modules/storage/pmem/pmem.py [hunk 5]
```diff
@@ -76,6 +76,39 @@
...
+          - The size of namespace. This option supports the suffixes C(k) or C(K) or C(KB) for KiB,
+            C(m) or C(M) or C(MB) for MiB, C(g) or C(G) or C(GB) for GiB and C(t) or C(T) or C(TB) for TiB.
+          - This option is required if multiple namespaces are configured.
+          - If this option is not set, all of the avaiable space of a region is configured.
+        type: str
+        required: false
+  namespace_append:
+    description:
+     - Enable to append the new namespaces to the system.
+     - The default is C(false) so the all existing namespaces not listed in I{namespace) are removed.
```
Comment (felixfontein) [inline plugins/modules/storage/pmem/pmem.py:108]:
```suggestion
     - The default is C(false) so the all existing namespaces not listed in I(namespace) are removed.
```

**Comment by mizumm** [comment]
Thank you for your review and suggestion! I woud be glad if this PR could be merged to 4.5.0.

**Review by felixfontein — COMMENTED** [review]
Besides that, looks good to me!

File: plugins/modules/storage/pmem/pmem.py (hunk 5 above)
Comment (mizumm) [reply plugins/modules/storage/pmem/pmem.py]: Thank you for poiting it out! I have applied the fix.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@mizumm thanks for your contribution!

**Pull request merged**